### PR TITLE
Increase comparison tolerance in SciPy spatial RotationMean subtest

### PR DESCRIPTION
**Pull request by andportnoy**
Previous value leads to failures on A100 runners in github.com/NVIDIA/JAX-Toolbox CI:
https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6144692887/job/16670611913#step:8:1014

The suspected reason is the use of TF32 math for matmuls: decorating the function with `@jax.default_matmul_precision("float32")` allows the test to pass. We thought it's better to loosen the tolerance but preserve the original execution mode.

The fully qualified test case is `tests/scipy_spatial_test.py::LaxBackedScipySpatialTransformTests::testRotationMean0`

### CODE REVIEW

**Comment by andportnoy** [comment]
cc @nouiz @zhenying-liu @DwarKapex @yhtang

**Review by jakevdp — APPROVED** [review]

File: tests/scipy_spatial_test.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def testRotationMean(self, shape, dtype, rng_weights):
     args_maker = lambda: (rng(shape, dtype), jnp.abs(rng(shape[0], dtype)) if rng_weights else None)
     jnp_fn = lambda q, w: jsp_Rotation.from_quat(q).mean(w).as_rotvec()
     np_fn = lambda q, w: osp_Rotation.from_quat(q).mean(w).as_rotvec().astype(dtype)  # HACK
-    tol = 5e-3 if jtu.device_under_test() == 'tpu' else 1e-4
```
Comment (hawkinsp) [inline tests/scipy_spatial_test.py:235]: I think it's probably better to just tighten the matmul precision.

Bonus: this would probably let us get rid of the TPU-specific code here, too, because TPU is likely imprecise for the same reason that Ampere is imprecise.

Comment (andportnoy) [reply tests/scipy_spatial_test.py:235]: > Bonus: this would probably let us get rid of the TPU-specific code here, too, because TPU is likely imprecise for the same reason that Ampere is imprecise.

Well it's a wash since I am already removing TPU-specific code here :)

> I think it's probably better to just tighten the matmul precision.

Not a strongly held opinion, but it felt wrong to artificially "enhance" precision for testing purposes while the default mode of operation "in the wild" is TF32. Explicit tolerances in the tests sort of annotate the ops with the expected accuracy.

On the other hand we end up not applying the level of stringency that's expected of hardware that doesn't use TF32.

BTW the distribution of max absolute/relative differences is quite wild if you vary the seed. At 5e-3 I ran 40 different seeds before I hit one that causes a failure. But then it means even that tolerance is not sufficient for some cases.

Comment (hawkinsp) [reply tests/scipy_spatial_test.py:235]: Yeah, no strong feelings I guess. I tend to opt for testing in highest precision since we're all more likely to agree on what that means.

Comment (andportnoy) [reply tests/scipy_spatial_test.py:235]: Happy to comply with that convention going forward!
